lib/network/connection_manager.go

```diff
@@ -69,7 +69,10 @@ func (c *ConnectionManager) GetConnection(address string) (client NetworkClient)
 }
 
 func (c *ConnectionManager) Start() {
-	go c.connectValidators()
+	c.log.Debug("starting to connect to validators", "validators", c.validators)
+	for _, v := range c.validators {
+		go c.connectingValidator(v)
```

Comment:
Is this for performance or for fixing racecondition? :)

Reply:
That's part of the cleaning up code :)
And it should (slighly) improve performance as well :)